graphqlapi/src/main/java/io/stargate/graphql/web/RequestToHeadersMapper.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.graphql.web;
+
+import java.util.*;
+import javax.servlet.http.HttpServletRequest;
+
+public class RequestToHeadersMapper {
+  public static Map<String, String> getAllHeaders(HttpServletRequest request) {
```

Comment:
This logic needs to be duplicated because of this:

In the stargate we are using two dependencies that are providing the `HttpServletRequest` class:
- `jakarta.servlet-api`
- `javax.servlet:javax.servlet-api`

I want to add logic to the core module that uses this class.
Unfortunately, it does not work with OSGIi; it fails with:
```INFO  [Exec Stream Pumper] 2021-01-08 11:12:04,337 StargateContainer.java:435 - sg0-0> ! java.lang.LinkageError: loader constraint violation: loader (instance of org/apache/felix/framework/BundleWiringImpl$BundleClassLoader) previously initiated loading for a different type with name "javax/servlet/http/HttpServletRequest"```
because the `HttpServletRequest` class is sometimes (for some modules) javax and for some others it is Jakarta. Does someone observe this problem?
Shouldn’t we use the same implementation for all modules? I need to use the same logic that uses `HttpServletApi` in `restapi`, `grapqhlapi`, and `auth-api`. I solved this problem by duplicating small part of the logic that uses `HttpServletRequest` but it is not ideal.

Reply:
We could use an embedded jar (cf. `persistence-common`) as opposed to an OSGi bundle in this case. The classes in the embedded jar will be resolved against the servlet API version specific to each containing bundle. That should be successful as long at the servlet API version is compatible.

Reply:
We should postpone it to a separate ticket since this one is a high priority, and this code duplication is not blocking.There is already ticket to capture that:
https://github.com/stargate/stargate/issues/239